components/dashboard/src/workspaces/ConnectToSSHModal.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import React, { useState } from "react";
+import Modal from "../components/Modal";
+import Tooltip from "../components/Tooltip";
+import copy from "../images/copy.svg";
+import TabMenuItem from "../components/TabMenuItem";
+
+function InputWithCopy(props: { value: string; tip?: string; className?: string }) {
+    const [copied, setCopied] = useState<boolean>(false);
+    const copyToClipboard = (text: string) => {
+        const el = document.createElement("textarea");
+        el.value = text;
+        document.body.appendChild(el);
+        el.select();
+        try {
+            document.execCommand("copy");
+        } finally {
+            document.body.removeChild(el);
+        }
+        setCopied(true);
+        setTimeout(() => setCopied(false), 2000);
+    };
+    const tip = props.tip ?? "Click to copy";
+    return (
+        <div className={`w-full relative ${props.className ?? ""}`}>
+            <input
+                disabled={true}
+                readOnly={true}
+                autoFocus
+                className="w-full pr-8 overscroll-none"
+                type="text"
+                defaultValue={props.value}
+            />
+            <div className="cursor-pointer" onClick={() => copyToClipboard(props.value)}>
+                <div className="absolute top-1/3 right-3">
+                    <Tooltip content={copied ? "Copied!" : tip}>
+                        <img src={copy} alt="copy icon" title={tip} />
+                    </Tooltip>
+                </div>
+            </div>
+        </div>
+    );
+}
+
+interface SSHProps {
+    workspaceId: string;
+    ownerToken: string;
+    ideUrl: string;
+}
+
+function SSHManualView(props: SSHProps) {
+    return (
+        <>
+            <div className="mt-1 mb-2">
+                <p className="text-gray-500 whitespace-normal text-base">
+                    The following args can be used with SSH to connect to this workspace.
+                </p>
+            </div>
+            <div className="flex flex-col">
+                <div className="mt-2">
+                    <h4>Host</h4>
+                    <InputWithCopy value={props.ideUrl} />
+                </div>
+                <div className="mt-2">
+                    <h4>Username</h4>
+                    <InputWithCopy value={props.workspaceId} />
+                </div>
+                <div className="mt-2">
+                    <h4>Password</h4>
+                    <InputWithCopy value={props.ownerToken} />
```

Comment:
**issue**: Printing the password here feels like we're exposing information we could better hide, no?